roles/test-utils/mining-device/src/main.rs

```diff
@@ -614,33 +613,40 @@ impl Miner {
 
 // returns hashrate based on how fast the device hashes over the given duration
 fn measure_hashrate(duration_secs: u64) -> f64 {
-    let mut share = generate_random_80_byte_array();
+    let mut rng = thread_rng();
+    let prev_hash: [u8; 32] = generate_random_32_byte_array().to_vec().try_into().unwrap();
```

Comment:
so we are using a random number for the prevhash?
why?

I assume prevhash can be anything on this context, since all shares will be discarded anyways.

but maybe we should add a comment for making this explicitly clear?

Reply:
better to use random so you are sure that there wont be optimization.